richreview/richreview.py

```diff
@@ -1,504 +0,0 @@
-"""This XBlock manages backend data flow of the RichReview web app."""
-import traceback
-import hashlib
-import json
-from uuid import uuid4
-
-from xblock.core import XBlock
-from xblock.fields import Scope, String, Dict
-from xblock.fragment import Fragment
-from djpyfs import djpyfs
-from django.conf import settings
-from webob.response import Response
-from boto.s3.connection import Location
-from pys3website import pys3website
-
-from util import *
-from osfs_utils import upload_webapp, clear_webapp, osfs_mkdir, osfs_copy_file, osfs_save_formfile
-from mupla_ctype import mupla_ctype
-
-RESOURCE_EXPIRATION_TIME = 7200 # 2 hours
-
-class RichReviewXBlock(XBlock):
-    """
-    This XBlock
-        (1) serves the RichReview web app at '../webapps/app_richreview', and
-        (2) manages backend data flow of the app.
-    """
-
-    # This filesystem stores the document (.PDF), the document metadata (.js), and audio (.WAV) files.
-    # The audio files will be stored at '<xblockid>/<groupid>/'.
-    # The document files will be stored at '<xblockid>/<docs>/'.
-    # Note that the PDF file will be renamed to the file contents' SHA1 hash value
-    #   (e.g., '6adfb183a4a2c94a2f92dab5ade762a47889a5a1.pdf').
-    fs = djpyfs.get_filesystem("rrfs")
-    # The RichReview web app will be served via this static URL.
-    # The web app files are stored in '/static/app', and will be maintained or updated through a github repository.
-    if hasattr(settings, 'DJFS'):
-        djfs_settings = settings.DJFS
-    else:
-        djfs_settings = {'type' : 'osfs',
-                         'directory_root' : 'common/static/djpyfs',
-                         'url_root' : '/static/djpyfs'}
-    if djfs_settings["type"] == "osfs":
-        clear_webapp(fs, 'richreview_web_app')
-        richreview_app_url =  upload_webapp(fs, "/webapps/richreview", "richreview_web_app", exclude=r'^\.')
-
-        clear_webapp(fs, 'multicolumn_web_app')
-        multicolumn_app_url =  upload_webapp(fs, "/webapps/multicolumn", "multicolumn_web_app", exclude=r'^\.')
```

Comment:
It's a bad idea to upload the app during file parse time. This should be a separate operation as part of deployment (or not needed at all, with the expectation that the XBlock runtime will make `public/webapp` accessible.